### PR TITLE
feat(FEC-11226): Client-Side CDN Switching using NPAW Smart Switch

### PR DESCRIPTION
**Pull request by dan-ziv**

### CODE REVIEW

**Comment by yairans** [comment]
@dan-ziv you still have async/await in the code

File: src/smart-switch-engine-decorator.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  get active(): boolean {
+    return this._active;
+  }
+
+  load(startTime: ?number): Promise<Object> {
+    return new Promise((resolve, reject) => {
+      this._logger.debug('load API');
+      (this._plugin.getCdnBalancerUrl() || Promise.resolve())
+        .then(cdnBalancerUrl => {
+          if (cdnBalancerUrl) {
```
Comment (yairans) [inline src/smart-switch-engine-decorator.js:32]: need else no? for when this._plugin.getCdnBalancerUrl() is null

File: src/smart-switch-engine-decorator.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            this._logger.debug('Set new engine src', cdnBalancerUrl);
+            this._engine.src = cdnBalancerUrl;
+          }
+        })
+        .catch(err => this._logger.warn(err.message))
+        .finally(() => {
```
Comment (yairans) [inline src/smart-switch-engine-decorator.js:38]: working on ie11?

File: src/smart-switch-engine-decorator.js (hunk 1 above)
Comment (dan-ziv) [reply src/smart-switch-engine-decorator.js:32]: No, the rest of the logic will be done in the finally

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          api_key: $NPM_TOKEN
+          email: $NPM_EMAIL
+          skip_cleanup: true
+          on:
+            tags: true
+            branch: master
+      after_deploy:
+        - currentVersion=$(npx -c 'echo "$npm_package_version"')
+        - chmod +x ./scripts/after_deploy.sh
+        - ./scripts/after_deploy.sh "youbora" "$currentVersion" "$JENKINS_TAG_TOKEN"
```
Comment (Yuvalke) [inline .travis.yml:58]: change to smartswitch

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        skip_cleanup: true
+        tag: canary
+        on:
+          tags: false
+          branch: master
+      after_deploy:
+        - currentVersion=$(npx -c 'echo "$npm_package_version"')
+        - echo $currentVersion
+        - chmod +x ./scripts/after_deploy.sh
+        - ./scripts/after_deploy.sh "youbora" "$currentVersion" "$JENKINS_CANARY_TOKEN"
```
Comment (Yuvalke) [inline .travis.yml:77]: same change from youbora to smartswitch

File: src/smart-switch-engine-decorator.js (hunk 2 above)
Comment (Yuvalke) [reply src/smart-switch-engine-decorator.js:38]: R u sure? I think u need a polyfill for finally

Comment (dan-ziv) [reply src/smart-switch-engine-decorator.js:38]: I'm sure.
You also have finally in playkit-js.

**Pull request merged**